### PR TITLE
[GLT-4063] allow adding libraries to requisitions

### PR DESCRIPTION
**Pull request by djcooke**
Jira ticket or GitHub issue: https://jira.oicr.on.ca/browse/GLT-4063

- [x] Includes a change file

### CODE REVIEW

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java [hunk 1]
```diff
@@ -242,6 +253,12 @@ public long create(Library library) throws IOException {
       } else {
         transferService.create(transfer);
       }
+      // Don't log initial additions to requisition, as that's part of the requisition creation
+      if (library.getRequisition() != null
+          && library.getRequisition().getCreationTime().toInstant()
+              .isBefore(Instant.now().minus(1, ChronoUnit.HOURS))) {
```
Comment (grapigeau) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java:259]: Is there a likely case where an addition will be made to a requisition within an hour that isn't the initial one?

Comment (djcooke) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java:259]: It's very likely, but that's close enough to consider it part of the creation - the person is most likely creating the requisition in multiple steps

**Review by grapigeau — APPROVED** [review]

File: miso-web/src/main/webapp/scripts/bulk.js [hunk 2]
```diff
@@ -775,13 +827,127 @@ BulkUtils = (function ($) {
...
+      requisition: function (dataProject) {
+        return [
+          {
+            title: "Requisition Alias",
+            data: "requisitionAlias",
+            type: "text",
+            maxLength: 150,
+            description:
+              "Should usually match the ID of a requisition form stored in a separate system. Enter an alias to" +
+              "search for existing requisitions.",
```
Comment (avarsava) [inline miso-web/src/main/webapp/scripts/bulk.js:839]: won't this render as 'tosearch'?

File: miso-web/src/main/webapp/scripts/bulk_library.js [hunk 3]
```diff
@@ -274,16 +286,15 @@ BulkTarget.library = (function ($) {
...
-          title: "Project",
-          type: "text",
-          data: Constants.isDetailedSample ? "projectCode" : "projectName",
-          include: !config.isLibraryReceipt,
-          disabled: true,
-        },
-        BulkUtils.columns.assay()
-      );
+      if (config.isLibraryReceipt) {
+      }
```
Comment (avarsava) [inline miso-web/src/main/webapp/scripts/bulk_library.js:290]: What is this empty if block for?

File: miso-web/src/main/webapp/scripts/bulk.js (hunk 2 above)
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/bulk.js:839]: Yep! fixed

File: miso-web/src/main/webapp/scripts/bulk_library.js (hunk 3 above)
Comment (djcooke) [reply miso-web/src/main/webapp/scripts/bulk_library.js]: Removed

**Review by avarsava — APPROVED** [review]

**Pull request merged**